nussl/datasets/hooks.py

```diff
@@ -417,6 +427,146 @@ def process_item(self, item):
                 "'mix', 'sources' as keys!")
         return output
 
+
+class SalientExcerptMixSourceFolder(OnTheFly):
+    """
+    Given a dataset in the expected format, this class identifies and
+    partitions the folder into a dataset containing segment_dur second audio
+    clips that contain a desired salient source. See MixSourceFolder for more
+    details on expected directory structure.
+
+    Args:
+        folder (str): Location that should be processed to produce the
+            list of files.
+        salient_src (str): The name of the source that will be used to identify
+         salient samples in the dataset. e.g. ('drums')
+        sample_rate (int, optional): the sampling rate for the audio files.
```

Comment:
Nit: Make sure to use proper grammar in docstrings, including capitalizing the first word of each sentence.